config/silabs/CMakeLists.txt

```
@@ -0,0 +1,157 @@
+#
+#   Copyright (c) 2024 Project CHIP Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+#
+#   @file
+#     CMake sub-project defining 'chip' target which represents CHIP library
+#     and other optional libraries like unit tests, built with 'nxp_zephyr'
+#     platform.
+#     Since CHIP doesn't provide native CMake support, ExternalProject
+#     module is used to build the required artifacts with GN meta-build
+#     system. It is assumed that find_package(Zephyr) has been called before
+#     including this file.
+#
+
+if (CONFIG_CHIP)
+
+include(ExternalProject)
+include(../zephyr/ota-image.cmake)
+include(../zephyr/zephyr-util.cmake)
+include(generate_factory_data.cmake)
+
+# ==============================================================================
+# Prepare CHIP configuration based on the project Kconfig configuration
+# ==============================================================================
+
+# Set paths
+
+if (NOT CHIP_ROOT)
+    get_filename_component(CHIP_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../.. REALPATH)
+endif()
+get_filename_component(GN_ROOT_TARGET ${CHIP_ROOT}/config/silabs/chip-gn REALPATH)
+
+include(${CHIP_ROOT}/config/silabs/cmake/common.cmake)
+
+# Prepare compiler flags
+if (CHIP_CFLAGS)
+    matter_add_flags("${CHIP_CFLAGS}")
+endif()
+
+matter_add_cxxflags(-D_POSIX_C_SOURCE=200809)
+
+if (CONFIG_ARM)
+    matter_add_cflags(--specs=nosys.specs)
+endif()
+
+if(CONFIG_MBEDTLS)
+    zephyr_include_directories($<TARGET_PROPERTY:mbedTLS,INTERFACE_INCLUDE_DIRECTORIES>)
+    zephyr_compile_definitions($<TARGET_PROPERTY:mbedTLS,INTERFACE_COMPILE_DEFINITIONS>)
+endif()
+
+zephyr_get_compile_flags(ZEPHYR_CFLAGS_C C)
+matter_add_cflags("${ZEPHYR_CFLAGS_C}")
+zephyr_get_compile_flags(ZEPHYR_CFLAGS_CC CXX)
+matter_add_cxxflags("${ZEPHYR_CFLAGS_CC}")
+zephyr_get_gnu_cpp_standard(ZEPHYR_GNU_CPP_STD)
+matter_add_cxxflags(${ZEPHYR_GNU_CPP_STD})
```

spacing for readability
```suggestion
zephyr_get_compile_flags(ZEPHYR_CFLAGS_C C)
matter_add_cflags("${ZEPHYR_CFLAGS_C}")

zephyr_get_compile_flags(ZEPHYR_CFLAGS_CC CXX)
matter_add_cxxflags("${ZEPHYR_CFLAGS_CC}")

zephyr_get_gnu_cpp_standard(ZEPHYR_GNU_CPP_STD)
matter_add_cxxflags(${ZEPHYR_GNU_CPP_STD})
```